app/models/dataset_record/darwin_core/taxon.rb

```diff
@@ -137,6 +137,10 @@ def import(dwc_data_attributes = {})
               original_combination_parents << next_parent
             end
 
+            # in cases where the taxon original combination is subgenus of self eg Sima (Sima), the first parent of the list
+            # should be dropped because it hasn't been imported yet
+            original_combination_parents = original_combination_parents.drop_while {|p| p.status != "Imported" }
```

Comment:
Are preconditions safe enough that using `drop_while` instead of `reject` is good? Or are you actually wanting to have possible tail imported records in the list?

Reply:
Originally, I chose `drop_while` so that it wouldn't have to keep iterating through the list once it found imported items. 

IIRC, only the first item in the list isn't imported, so a non-imported element somewhere else is a symptom of an unrelated issue. In that case, I don't mind it breaking to alert us.

But yes, the preconditions are very safe on this one, since the parents are added in hierarchical order.